python2/src/countMVS.py

```diff
@@ -1148,12 +1148,14 @@ def _resolve_hostnames_to_ips(self):
         logging.info('Attempting to resolve hostnames to ips')
         for machine_identifier, log_sources in self.mvs_results.get_device_map().items():
             logging.info('Attempting to resolve machine identifier %s to an ip address', machine_identifier)
-            device_ip = IPParser.get_device_ip(machine_identifier)
-            if not device_ip:
-                logging.info('Unable to resolve machine identifier %s to an ip address', machine_identifier)
+            try:
+                device_ip = IPParser.get_device_ip(machine_identifier)
+            except Exception as err:
+                logging.info('Unable to resolve machine identifier %s to an ip address. Error %s', machine_identifier,
+                             str(err))
                 continue
-            if device_ip != machine_identifier:
-                logging.info('Resolved machine identifer %s to ip address %s', machine_identifier, device_ip)
+            if device_ip is not None and device_ip != machine_identifier:
```

Comment:
I don't think you need the `device_ip is not None` check. The comparison `device_ip != machine_identifier` will work whatever the value of `device_ip`.

Reply:
You're right, I'm talking rubbish, lines 1158/9 musn't execute if device_ip is empty.

You could simplify to this which is more Pythonic: `if device_ip and device_ip != machine_identifier`

Reply:
No probelm that makes sense I will do that here now